firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -27,6 +28,7 @@
 import com.google.firebase.FirebaseException;
 import com.google.firebase.FirebaseOptions;
 import com.google.firebase.heartbeatinfo.HeartBeatInfo;
+import com.google.firebase.iid.FirebaseInstanceId;
```

Comment:
How can this work if there is no Instance ID available?

Reply:
Check the isIIDVersionCompatible(). If there is no Instance ID it catches NoClassDefFoundError and returns compatibility true.